src/main/java/org/mockbukkit/mockbukkit/inventory/InventoryViewMock.java

```diff
@@ -145,14 +145,52 @@ public void setCursor(@Nullable ItemStack item)
 	@Override
 	public void setItem(int slot, @Nullable ItemStack item)
 	{
-		// TODO Auto-generated method stub
+		int topSize = getTopInventory().getSize();
+		if (slot < 0)
+		{
+			// TODO : ???
+			throw new UnimplementedOperationException();
```

Comment:
Just from a hunch, this should probably throw an IllegalArgumentException.

Reply:
Just tested: it returns null

Reply:
This is weird: -1 drops the itemstack
```java
    @EventHandler
    public void onInvOpen(InventoryOpenEvent event) {
        event.getView().setItem(-1, new ItemStack(Material.DIAMOND_SWORD));
    }
```

Reply:
@Thorinwasher : I would need some guidance here...

What do I do?
```java
			if (item != null) {
				Location location = topInventory.getLocation();
				if (location != null) {
					// Drop the item in the world at the inventory's location
					location.getWorld().dropItem(location, item);
				}
			}
```

But the getLocation() for the inventory isn't implemented. And if I start that, it's going to get me down the rabbit hole.
What do I do?
I'd suggest to add a comment refering this PR + what needs to be done, but still throw `UnimplementedOperationException`?

Reply:
> Thats hillariously wrong. Let's just throw an index out of bounds exception then, as this is unintended behavior

Maybe, but it's what MC does...
So we should implement it as closely as possible in my opinion.

My 2cts

Reply:
Actually, it's correct behaviour:
https://hub.spigotmc.org/javadocs/spigot/org/bukkit/inventory/InventoryView.html#setItem(int,org.bukkit.inventory.ItemStack)

However, they state "it is not required"

Reply:
I totally see your point of view there. But don't forget mockbukkit is to test plugins. For debugging purposes, throwing an exception would help more